pytensor/link/pytorch/dispatch/subtensor.py

```diff
@@ -0,0 +1,78 @@
+from pytensor.link.pytorch.dispatch.basic import pytorch_funcify
+from pytensor.tensor.subtensor import (
+    AdvancedIncSubtensor1,
+    AdvancedSubtensor,
+    AdvancedSubtensor1,
+    IncSubtensor,
+    Subtensor,
+    indices_from_subtensor,
+)
+from pytensor.tensor.type_other import MakeSlice
+
+
+@pytorch_funcify.register(Subtensor)
+def pytorch_funcify_Subtensor(op, node, **kwargs):
+    idx_list = getattr(op, "idx_list", None)
+
+    def subtensor(x, *ilists):
+        indices = indices_from_subtensor(ilists, idx_list)
+        for i in indices:
+            if isinstance(i, slice):
+                if i.step and i.step < 0:
+                    raise NotImplementedError(
+                        "Negative step sizes are not supported in Pytorch"
+                    )
+
+        return x[indices]
+
+    return subtensor
+
+
+@pytorch_funcify.register(AdvancedSubtensor1)
+@pytorch_funcify.register(AdvancedSubtensor)
+def pytorch_funcify_AdvSubtensor(op, node, **kwargs):
+    def advsubtensor(x, *indices):
+        for i in indices:
+            if isinstance(i, slice):
+                if i.step and i.step < 0:
+                    raise NotImplementedError(
+                        "Negative step sizes are not supported in Pytorch"
+                    )
+        return x[indices]
+
+    return advsubtensor
+
+
+@pytorch_funcify.register(MakeSlice)
+def pytorch_funcify_makeslice(op, **kwargs):
+    def makeslice(*x):
+        return slice(x)
+
+    return makeslice
+
+
+@pytorch_funcify.register(IncSubtensor)
+@pytorch_funcify.register(AdvancedIncSubtensor1)
```

Comment:
Split IncSubtensor (basic indexing) and AdvancedIncSubtensor (advanced indexing) like you did for Subtensor and AdvancedSubtensor. Also why is AdvavcedSubtensor (without 1) not covered?